packages/jellyfish-api-core/__tests__/category/poolpair/poolswap.test.ts

```diff
@@ -0,0 +1,230 @@
+import { MasterNodeRegTestContainer } from '@defichain/testcontainers'
+import { ContainerAdapterClient } from '../../container_adapter_client'
+import BigNumber from 'bignumber.js'
+import {
+  addPoolLiquidity,
+  createPoolPair,
+  createToken,
+  getNewAddress,
+  mintTokens,
+  utxosToAccount
+} from '@defichain/testing'
+import { RpcApiError } from '../../../src'
+import { PoolSwapMetadata, PoolPairInfo } from '@defichain/jellyfish-api-core/category/poolpair'
+
+describe('poolSwap', () => {
+  const container = new MasterNodeRegTestContainer()
+  const client = new ContainerAdapterClient(container)
+
+  beforeAll(async () => {
+    await container.start()
+    await container.waitForReady()
+    await container.waitForWalletCoinbaseMaturity()
+  })
+
+  afterAll(async () => {
+    await container.stop()
+  })
+
+  it('should poolSwap with UTXOS ', async () => {
+    const tokenAddress = await getNewAddress(container)
+    const dfiAddress = await getNewAddress(container)
+    const poolLiquidityAddress = await getNewAddress(container)
```

Comment:
you can use `getaccount` to check the after poolswap balance changes, its a guidance for people know how to manage their coin in each account

Reply:
I have added the accountReceiver balance check to `poolSwap` block. I think that will suffice.